src/main/java/org/example/bookstore/exception/CustomGlobalExceptionHandler.java

```diff
@@ -0,0 +1,71 @@
+package org.example.bookstore.exception;
+
+import java.time.LocalDate;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDate.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult()
+                .getAllErrors().stream()
+                .map(this::getErrorMassage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    private String getErrorMassage(ObjectError e) {
+        if (e instanceof FieldError) {
+            String field = ((FieldError) e).getField();
+            String message = e.getDefaultMessage();
+            return field + " " + message;
+        }
+        return e.getDefaultMessage();
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException ex) {
+        HttpStatus status = HttpStatus.NOT_FOUND;
+        Map<String, Object> body = createErrorMessageBody(status, ex.getMessage());
+        return new ResponseEntity<>(body, status);
+    }
+
+    @ExceptionHandler(InvalidPriceException.class)
+    protected ResponseEntity<Object> handleInvalidPriceException(InvalidPriceException ex) {
+        HttpStatus status = HttpStatus.BAD_REQUEST;
+        Map<String, Object> body = createErrorMessageBody(status, ex.getMessage());
+        return new ResponseEntity<>(body, status);
+    }
```

Comment:
duplicate code, we will remove the price ex in that solution, but for the future - if you have the same logic of handling some exceptions just have a method like that
```java
    @ExceptionHandler({
        CustomEx1.class,
        CustomEx2.class})
    public ResponseEntity<Object> handleBadRequestException(RuntimeException ex) {
        HttpStatus status = HttpStatus.BAD_REQUEST;
        var body = createErrorMessageBody(status, ex.getMessage());
        return new ResponseEntity<>(body, status);
    }
```